src/main/scala/dotty/dokka/ScalaPageCreator.scala

```diff
@@ -154,13 +154,63 @@ class ScalaPageCreator(
         modifyContentGroup(content, modifiedContent)
     }
 
+    def insertEnumTab(clazz: DClass, defContent: ContentGroup): ContentGroup = {
+        val content = getContentGroupWithParents(defContent, p => p.getStyle.asScala.contains(ContentStyle.TabbedContent))
+        val addedContent = PageContentBuilder(commentsToContentConverter, signatureProvider, logger).contentFor(clazz)(builder => {
+            groupingBlock(
+                builder,
+                "Entries",
+                List(() -> clazz.get(EnumExtension).enumEntries.sortBy(_.getName).toList),
+                (builder, _) => {
+                    kotlin.Unit.INSTANCE
+                },
+                (builder, elem) => {
+                    link(builder, elem.getName, elem.getDri)(kind = ContentKind.Main)
+                    sourceSetDependentHint(builder)( builder =>
+                        {
+                            contentForBrief(builder, elem)
+                            builder.unaryPlus(builder.buildSignature(elem))
+                            kotlin.Unit.INSTANCE
+                        },
+                        dri = Set(elem.getDri).asJava,
+                        sourceSets = elem.getSourceSets,
+                        kind = ContentKind.SourceSetDependentHint,
+                        styles = Set().asJava
+                    )
+                    kotlin.Unit.INSTANCE
+                }
+            )(
+                kind = ContentKind.Main,
+                sourceSets = builder.getMainSourcesetData.asScala.toSet,
+                styles = Set(),
+                extra = PropertyContainer.Companion.empty().plus(SimpleAttr.Companion.header("Entries")),
+                false,
+                true,
+                Nil,
+                false,
+                true
+            )
+            kotlin.Unit.INSTANCE
+        })
+
+        val modifiedContent = content(0).copy(
+            (content(0).getChildren.asScala ++ List(addedContent)).asJava,
+            content(0).getDci,
+            content(0).getSourceSets,
+            content(0).getStyle,
+            content(0).getExtra
+        )
+        modifyContentGroup(content, modifiedContent)
+    }
+
     override def contentForClasslike(c: DClasslike): ContentGroup = {
         val defaultContent = super.contentForClasslike(c)
       
         c match{
             case clazz: DClass =>
                 val op1 = insertCompanion(clazz, defaultContent)
-                insertCustomExtensionTab(clazz, op1)
+                val op2 = insertCustomExtensionTab(clazz, op1)
```

Comment:
I think we should rename `op1` and `op2` into more meaningfull names (`withCompanion` and `withClass`)

Reply:
renamed